### DISCUSSION TITLE
Role Limiter

### DISCUSSION THREAD

**Discussion by ghost** [Kilvoctu/aiyabot#96] [Ideas]
Hi! Is it possible to add a limiter? To limit how many times people can use the /draw command per-day according to a configuration for their specific role. Basically to stop mass amounts of spam unless their role permits it.

**Comment by Kilvoctu** [comment]
Sure, it's possible, 🤔 once I figure out how to do it. I'll look into it and try to find some tutorials on the subject.  
Very reasonable idea; I know how intensive SD can be lol

**Reply by ghost** [reply]
Sweet; can't wait to see what you implement! :D

**Comment by Kilvoctu** [comment]
I'm probably gonna have to put this on hold. Pycord doesn't give any great way to do ratelimiting in a way that makes sense for this use case. I tried a few implementations, but this seems like something that would have to be done from scratch.  
If you want to, you can check out https://github.com/Kilvoctu/aiyabot/tree/ratelimit, and try to work with that.


Also, I did check out your Discord server and I see you've removed all attribution to me and my work in your distribution. Please note this is a [licensed](https://github.com/Kilvoctu/aiyabot/blob/main/LICENSE) project, and you're breaking each condition of the license... I'd like to not report a license violation 😩  
I don't ask for much, just credit me for my work. If you could at least return the "About me" button and link back to this repo, then after that you can fulfill the other license conditions.

**Comment by ghost** [comment]
The about me button was gonna be returned in due time anyway as "Credits' instead of "About Me" I wanted to include multiple credits to both Aiyabot & stable diffusion webui. I'll get it back real quick here including all the other credits I wanted to add.

Don't mind me asking but which other conditions am I breaking?

**Reply by Kilvoctu** [reply]
Your distribution should be open source too and include the same license. And people should know how it was changed. Your github profile was made private at some point, so idunno if the source code is elsewhere or what. 

You can check this fork https://github.com/killfrenzy96/AiyaFrenzyBot for an exceptional version of AIYA and how they handle it. The FrenzyBot is massively divergent from my own.
